enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,877 @@
+
+- For headers like X-Forwarded-For it makes sense to add append action. So why does this enhancement not cover append ?
+  Answer: Currently, we are proceeding with Set and Delete only. The epic demands these two only. We have confirmed with the stakeholders as well.
+  Talking concretely about the X-Forwarded headers, we have a dedicated IngressController's and Route field to handle them which does have an `Append` action:
+
+```sh
+$ oc explain ingresscontroller.spec.httpHeaders.forwardedHeaderPolicy
+KIND:     IngressController
+VERSION:  operator.openshift.io/v1
+
+FIELD:    forwardedHeaderPolicy <string>
+
+    DESCRIPTION:
+    forwardedHeaderPolicy specifies when and how the IngressController sets the
+    Forwarded, X-Forwarded-For, X-Forwarded-Host, X-Forwarded-Port,
+    X-Forwarded-Proto, and X-Forwarded-Proto-Version HTTP headers. The value
+    may be one of the following:
+    * "Append", which specifies that the IngressController appends the headers,
+    preserving existing headers.
+    * "Replace", which specifies that the IngressController sets the headers,
+    replacing any existing Forwarded or X-Forwarded-* headers.
+    * "IfNone", which specifies that the IngressController sets the headers if
+    they are not already set.
+    * "Never", which specifies that the IngressController never sets the
+    headers, preserving any existing headers. By default, the policy is
+    "Append".
+```
+
+- Would some headers be blocked for setting and deleting ?
+  Answer: Yes, this enhancement won't allow the user to set or delete HSTS headers. We already have a way to set this which is via route 
+  annotation. Providing one more way to add this can cause issues.
+
+- Many applications serve mixed content: static assets (images, js, css) and dynamic content (html, json). They usually want to cache the static assets on the client and do not care about session stickiness for those requests. Only the dynamic requests need to stick to the same server (session sticky).
+Most applications will set something like this for static assets:
+```sh
+cache-control: max-age=7200, must-revalidate, public
+```
+But the OpenShift router (HAProxy) will add cache-control: private which results in both headers in the response:
+```sh
+cache-control: private
+cache-control: max-age=7200, must-revalidate, public
+```
+This leads to issues with caching: CDNs like Cloudflare will not cache your files.
+Can we please make sure to only add the cache-control: private if there is no cache-control: public already present in the response ?
+Answer: A common technique is to put static assets on a different virtual host. In OpenShift terms, that means using a distinct route for static assets. This route can be annotated with haproxy.router.openshift.io/disable_cookies=true to prevent setting the cookie and thus prevent setting cache-control.
```

Comment:
```suggestion
Can we please make sure to only add the `cache-control: private` header if there is no `cache-control: public` header already present in the response?
Answer: A common technique is to put static assets on a different virtual host. In OpenShift terms, that means using a distinct route for static assets. This route can be annotated with `haproxy.router.openshift.io/disable_cookies=true` to prevent setting the cookie and thus prevent setting cache-control.
```

Reply:
done